python/cudf/cudf/tests/test_string.py

```diff
@@ -887,19 +884,17 @@ def test_string_join_key(str_data, str_data_raise, num_keys, how):
     pdf2 = pdf.copy()
     gdf2 = gdf.copy()
 
-    expectation = raise_builder(
-        [0 if how == "right" else str_data_raise], (AssertionError)
-    )
+    expect = pdf.merge(pdf2, on=list(range(num_keys)), how=how)
+    got = gdf.merge(gdf2, on=list(range(num_keys)), how=how)
 
-    with expectation:
-        expect = pdf.merge(pdf2, on=list(range(num_keys)), how=how)
-        got = gdf.merge(gdf2, on=list(range(num_keys)), how=how)
+    if len(expect) == 0 and len(got) == 0:
+        expect = expect.reset_index(drop=True)
+        got = got[expect.columns]  # reorder columns
```

Comment:
if I squint I think this is just a case that `assert_join_results_equal` can handle right?

Reply:
We only reorder in `assert_join_results_equal` if we're doing a right join, whereas here it looks like we want to reorder in every case. I'm not exactly sure why this test is written the way it is - just wanted to maintain the same semantics.